virttest/utils_backup.py

```diff
@@ -0,0 +1,285 @@
+    summary = 0
+    for entry in json_data:
+        if entry["data"]:
+            summary += entry["length"]
+    return summary/1024/1024
+
+
+def get_img_data_map(img_path, driver='qcow2'):
+    """
+    Get the data map of a image
+
+    :prama img_path: The path to the image
+    :return: The data map of the image
+    """
+    cmd = "qemu-img map -f %s %s --output=json -U" % (driver, img_path)
+    result = process.run(cmd, shell=True).stdout_text
+    json_data = json.loads(result)
+    data_map = []
+    for entry in json_data:
+        if entry["data"]:
+            data_map.append(entry)
+    return data_map
+
+
+def dd_data_to_vm_disk(vm, disk, bs, seek, count):
+    """
+    Generate some random data to a vm disk
+
+    :param vm: The vm we'll use
+    :param disk: The disk of the vm we'll use
+    :param bs: The 'bs' param of the 'dd' command
+    :param seek: The 'seek' param of the 'dd' command
+    :param count: The 'count' param of the 'dd' command
+    """
+    session = vm.wait_for_login()
+    dd_cmd = "dd if=/dev/urandom of=%s bs=%s seek=%s count=%s; sync"
+    dd_cmd %= (disk, bs, seek, count)
+    output = session.cmd_output(dd_cmd).strip()
+    logging.debug("Using dd to generate data to %s: %s", disk, output)
+    session.close()
+
+
+def dump_data_to_file(data_map, source_file_path, target_file_path, driver='qcow2'):
+    """
+    Dump real data from the source file to a target file according to data map
+
+    :param data_map: The data map used to indicate which part of the image
+    containing real data
+    :param source_file_path: The path to the source file
+    :param target_file_path: The path to the target file
+    :param driver: The image format of the source file
+    """
+    with open(target_file_path, 'w'):
+        cmd = "qemu-io -f %s -c \"read -vC %s %s\" %s -r -U  | sed '$d'"
+        for entry in data_map:
+            cmd %= (driver, entry["start"], entry["length"], source_file_path)
+            cmd += ">> %s" % target_file_path
+            process.run(cmd, shell=True)
+
+
+def cmp_backup_data(original_file, backup_file,
+                    original_file_driver='qcow2', backup_file_driver='qcow2'):
+    """
+    Compare the backuped data and the original data
+
+    :param original_file: The file containing the original data
+    :param backup_file: The file containing the backuped data
+    :param original_file_driver: Original image format
+    :param backup_file_driver: Backup image format
+    :return: Trun if data is correct, false if not
+    """
+    original_file_dump = original_file + ".dump"
+    backup_file_dump = backup_file + ".dump"
+    data_map = get_img_data_map(backup_file, backup_file_driver)
+    dump_data_to_file(data_map, original_file, original_file_dump, original_file_driver)
+    dump_data_to_file(data_map, backup_file, backup_file_dump, backup_file_driver)
+    if filecmp.cmp(original_file_dump, backup_file_dump):
+        return True
+    return False
```

Comment:
It's redundant, would you please remove it?

Reply:
will remove